test/integration/text_layer_spec.mjs

```diff
@@ -98,111 +99,270 @@ describe("Text layer", () => {
+                  /path through the program .*Hence, recording a/s
+                );
+
+              await moveInSteps(page, positionStartPage2, positionEndPage2, 20);
+              await page.mouse.up();
+
+              await expectAsync(page)
+                .withContext(`In ${browserName}, second selection`)
+                .toHaveRoughlySelected(
+                  /path through.*Hence, recording and .* tracing, and give/s
+                );
+            })
+          );
+        });
       });
 
-      it("doesn't jump when hovering on an empty area (multi-page)", async () => {
-        await Promise.all(
-          pages.map(async ([browserName, page]) => {
-            const scrollTarget = await getSpanRectFromText(
-              page,
-              1,
-              "Unlike method-based dynamic compilers, our dynamic com-"
-            );
-            await page.evaluate(top => {
-              document.getElementById("viewerContainer").scrollTop = top;
-            }, scrollTarget.y - 50);
-
-            const [
-              positionStartPage1,
-              positionEndPage1,
-              positionStartPage2,
-              positionEndPage2,
-            ] = await Promise.all([
-              getSpanRectFromText(
-                page,
-                1,
-                "Each compiled trace covers one path through the program with"
-              ).then(middlePosition),
-              getSpanRectFromText(
-                page,
-                1,
-                "or that the same types will occur in subsequent loop iterations."
-              ).then(middlePosition),
-              getSpanRectFromText(
-                page,
-                2,
-                "Hence, recording and compiling a trace"
-              ).then(middlePosition),
-              getSpanRectFromText(
-                page,
-                2,
-                "cache. Alternatively, the VM could simply stop tracing, and give up"
-              ).then(belowEndPosition),
-            ]);
+      describe("when selecting over a link", () => {
+        let pages;
+
+        beforeAll(async () => {
+          pages = await loadAndWait(
+            "annotation-link-text-popup.pdf",
+            `.page[data-page-number = "1"] .endOfContent`
+          );
+        });
+        afterAll(async () => {
+          await closePages(pages);
+        });
+        afterEach(() =>
+          Promise.all(
+            pages.map(([_, page]) =>
+              page.evaluate(() => window.getSelection().removeAllRanges())
+            )
+          )
+        );
 
-            await page.mouse.move(positionStartPage1.x, positionStartPage1.y);
-            await page.mouse.down();
+        function waitForClick(page, selector, timeout) {
+          return page.evaluateHandle(
+            (sel, timeoutDelay) => {
+              const element = document.querySelector(sel);
+              const timeoutSignal = AbortSignal.timeout(timeoutDelay);
```

Comment:
*This doesn't require any changes, but just an observation for future reference:*

According to MDN this feature is newer than the minimum browsers that we support, compare https://developer.mozilla.org/en-US/docs/Web/API/AbortSignal/timeout_static#browser_compatibility and https://github.com/mozilla/pdf.js/wiki/Frequently-Asked-Questions#faq-support

However, given that this is test-only code this seems totally fine since I don't believe that we've ever said anything about the test-suite(s) being fully runnable in older browsers.